non_iid_detection/requirements.txt

```diff
@@ -0,0 +1,7 @@
+# This notebook currently requires the latest developer version of cleanlab: pip install git+https://github.com/cleanlab/cleanlab.git 
```

Comment:
I think we can now remove this statement right? @elisno

Reply:
add `datasets` package as a dependency. Let's always assume user just has base version of cleanlab installed (but no optional dependencies installed).

Reply:
Why not `cleanlab[datalab]`?

Reply:
Removed it and updated with some requirements that I needed.

Reply:
because none of our examples notebooks tell user to install cleanlab. We always assume cleanlab is already installed. It imposes burden on developers to determine which versions of cleanlab are installed, dealing w different versions, etc